tests/unit/gtBSMetadata.cpp

```diff
@@ -0,0 +1,169 @@
+// * Neither the name  of Intel Corporation  nor the names of its contributors
+//   may be used to  endorse or promote  products derived  from this  software
+//   without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+// IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+// LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+// CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+// SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+// INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+// CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#ifdef __cplusplus
+
+extern "C" {
+#endif
+#include <opae/enum.h>
+#include <opae/properties.h>
+#include "bitstream_int.h"
+
+#ifdef __cplusplus
+}
+#endif
+
+#include "common_test.h"
+#include "gtest/gtest.h"
+#include "types_int.h"
+
+#define DECLARE_GUID(var, ...) uint8_t var[16] = {__VA_ARGS__};
+
+using namespace common_test;
+
+/**
+* @test    bs_metadata_01
+* @brief   Tests: read_gbs_metadata 
+* @details read_gbs_metadata returns BS metadata
+*          Then the return value is FPGA_OK<br>
+*/
+TEST(LibopaecBSMetadataCommonMOCKHW, bs_metadata_01) {
+
+	// Invalid input
+	fpga_result result = read_gbs_metadata(NULL, NULL);
+	ASSERT_NE(result, FPGA_OK);
+
+	// Invalid input
+	struct gbs_metadata gbs_metadata ;
+	result = read_gbs_metadata(NULL, &gbs_metadata);
+	ASSERT_NE(result, FPGA_OK);
+
+	// Invalid input bitstream
+	uint8_t bitstream[10] = "abcd";
+	result = read_gbs_metadata(bitstream, NULL);
+	ASSERT_NE(result, FPGA_OK);
+
+	// Invalid input bitstream
+	result = read_gbs_metadata(bitstream, &gbs_metadata);
+	ASSERT_NE(result, FPGA_OK);
+
+	// Invalid input bitstream
+	uint8_t bitstream_guid[] = "XeonFPGA·GBSv001S";
+	result = read_gbs_metadata(bitstream_guid, &gbs_metadata);
+	ASSERT_NE(result, FPGA_OK);
+
+	// Invalid input bitstream
+	uint8_t bitstream_guid_invalid1[] = "XeonFPGA·GBSv001";
+	result = read_gbs_metadata(bitstream_guid_invalid1, &gbs_metadata);
+	ASSERT_NE(result, FPGA_OK);
+
+	// Invalid input bitstream
+	uint8_t bitstream_guid_invalid2[] = "XeonFPGA·GBSv001S  {\"version/\": 640, \"afu-image\":  \
+		{\"clock-frequency-high\": 312, \"clock-frequency-low\": 156,\
+		\"power\": 50, \"interface-uuid\": \"1a422218-6dba-448e-b302-425cbcde1406\", \
+		\"magic-no\": 488605312, \"accelerator-clusters\"\
+		: [{\"total-contexts\": 1, \"name\": \"nlb_400\", \"accelerator-type-uuid\": \
+		\"d8424dc4-a4a3-c413-f89e-433683f9040b\"}]}, \"platform-name\": \"MCP\"}";

```

Comment:
Seems like you have non-alphanumeric byte(s)  at the end of the string. You might also want to consider using C++ raw strings (assuming gtapi is built with c++11).
https://en.cppreference.com/w/cpp/language/string_literal